src/gluonts/ext/hierarchicalforecast/_predictor.py

```diff
@@ -0,0 +1,316 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import List, Optional, Union, Any, Dict
+
+import numpy as np
+import pandas as pd
+
+from statsforecast import StatsForecast
+from hierarchicalforecast.core import HierarchicalReconciliation
+from hierarchicalforecast.core import _build_fn_name
+
+from gluonts.core.component import validated
+from gluonts.dataset import DataEntry
+from gluonts.dataset.util import forecast_start
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.model.forecast import QuantileForecast
+from gluonts.ext.statsforecast import ModelConfig
+
+models_without_fitted_capability = [
+    "ADIDA",
+    "CrostonClassic",
+    "CrostonOptimized",
+    "CrostonSBA",
+    "IMAPA",
+    "SeasWA",
+    "TSB",
+    "WindowAverage",
+]
+
+
+def get_formatted_S(
+    _S: Union[List[List[int]], np.ndarray],
+    ts_names: List[str],
+) -> pd.DataFrame:
+    """
+    We format the summation matrix S as a dataframe,
+    where the index and columns have the
+    corresponding time series names.
+    """
+
+    S = np.array(_S)
+
+    return pd.DataFrame(
+        S, index=ts_names, columns=ts_names[S.shape[0] - S.shape[1] :]
+    )
+
+
+def format_data_entry(entry: DataEntry, S: pd.DataFrame) -> pd.DataFrame:
+    """
+    formats data entry as required by hierarchicalforecast.
+
+    entry is a dictionary with keys: 'start', 'item_id', 'target'.
+    entry['target'] is a np.narray with shape (num_ts, num_timestamps),
+    and each row corresponds to one time series of the hierarchy.
+    The goal is to reshape this DataEntry as a dataframe where:
+    1) the index corresponds to the name of the time series,
+    2) the columns 'ds'/'y' correspond to timestamps/actuals, respectively.
+    """
+
+    df = pd.DataFrame(entry["target"]).T
+    df.columns = S.index.tolist()
+    df.index = pd.date_range(
+        start=entry["start"].start_time,
+        periods=df.shape[0],
+        freq=entry["start"].freq,
+    )
+
+    df = unpivot(df)
+    df = df.set_index("unique_id")
+
+    return df
+
+
+def unpivot(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    unpivots data frame
+
+    The input dataframe has as index the time stamps
+    and columns per time series of the hierarchy.
+    We unpivot this so that the final dataframe has
+    three columns, i.e. "unique_id", "ds", and "y", where
+    1) "unique_id" has the name of the corresponding time series,
+    2) "ds" has the corresponding time stamps,
+    3) "y" has the actuals.
```

Comment:
```suggestion
    Unpivot data frame.

    The input dataframe has as index the time stamps,
    and one column per each time series of the hierarchy.
    We unpivot this so that the final dataframe has
    three columns, i.e. ``"unique_id"``, ``"ds"``, and ``"y"``, where
    1) ``"unique_id"`` has the name of the corresponding time series,
    2) ``"ds"`` has the corresponding time stamps,
    3) ``"y"`` has the actuals.
```

Reply:
Thank you! I updated the PR with it.